### PR TITLE
add developer mode

### PR DESCRIPTION
**Pull request by IrinaShkviro**
Initial commit for add developer mode. Developer mode enables debug menu with menu for crash testing inside for now. Menu is hidden by default, but becomes visible when Orbit is started with --devmode flag.

Only one crash option is added for now (CHECK(false)).

### CODE REVIEW

File: OrbitQt/orbitmainwindow.cpp [hunk 1]
```diff
@@ -737,3 +742,8 @@ void OrbitMainWindow::on_actionOutputDebugString_triggered(bool checked) {
 void OrbitMainWindow::on_actionUploadDumpsToServer_triggered(bool checked) {
   GOrbitApp->EnableUploadDumpsToServer(checked);
 }
+
+//-----------------------------------------------------------------------------
+void OrbitMainWindow::on_actionCheckFalse_triggered() {
```
Comment (pierricgimmig) [inline OrbitQt/orbitmainwindow.cpp:747]: CHECK generates a crashdump right?  We could potentially also add a menu entry for a null pointer dereference, not necessarily in this PR.

File: OrbitQt/main.cpp [hunk 2]
```diff
@@ -33,6 +33,9 @@ ABSL_FLAG(bool, enable_stale_features, false,
           "Enable obsolete features that are not working or are not "
           "implemented in the client's UI");
 
+ABSL_FLAG(bool, enable_debug_menu, false,
```
[the remark this answers is not in the document]
Comment (IrinaShkviro) [reply OrbitQt/main.cpp]: Yes, it much more better name. Thank you!

File: OrbitQt/orbitmainwindow.cpp (hunk 1 above)
Comment (IrinaShkviro) [reply OrbitQt/orbitmainwindow.cpp:747]: Yes, I want to add several types of checks in one of the next PRs.

**Comment by IrinaShkviro** [comment]
Thank you for the review!

File: OrbitQt/orbitmainwindow.cpp (hunk 1 above)
Comment (alenl-google) [reply OrbitQt/orbitmainwindow.cpp:747]: Yes, this is exactly what I meant. I don't see a reason to test the upload without going through the full workflow (app causes an access violation->crashpad process activates, etc)
Two most common cases that need to be tested are a null pointer access and stack overflow. Since stack overflow is quite different in terms of how it is caught and how the stacks need to be collected and decoded, this is also important to test.
